### PR TITLE
[thci] fix RA configuration for eth0 in the host device

### PR DESCRIPTION
**Pull request by edmont**
`accept_ra` was not being set properly for host reference device.

Some cleanup is added for simplificaiton.

### CODE REVIEW

File: tools/harness-thci/OpenThread_BR.py [hunk 1]
```diff
@@ -326,10 +324,14 @@ def _deviceBeforeReset(self):
     def _deviceAfterReset(self):
```
Comment (simonlingoogle) [inline tools/harness-thci/OpenThread_BR.py:324]: Do we always call `_deviceAfterReset` even for the first test?

Comment (edmont) [reply tools/harness-thci/OpenThread_BR.py:324]: Yes, `reset` is being called when adding devices to the setup.

Comment (simonlingoogle) [reply tools/harness-thci/OpenThread_BR.py:324]: SGTM

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**